jina/executors/indexers/cache.py

```diff
@@ -67,19 +66,18 @@ def __init__(self, index_filename: Optional[str] = None, field: Optional[str] =
         if self.field not in self.supported_fields:
             raise ValueError(f"Field '{self.field}' not in supported list of {self.supported_fields}")
 
-    def add(self, doc_id: 'UniqueId', *args, **kwargs):
+    def add(self, doc_id: str, *args, **kwargs):
         """Add a document to the cache depending on `self.field`.
 
         :param doc_id: document id to be added
         """
-        self.query_handler.ids.append(doc_id)
-
         # optimization. don't duplicate ids
         if self.field != ID_KEY:
             data = kwargs.get(DATA_FIELD, None)
-            if data is None:
-                raise ValueError(f'Got None from CacheDriver')
-            self.query_handler.content_hash.append(data)
+        else:
+            data = doc_id
+        self.query_handler.id_to_cache_val[doc_id] = data
```

Comment:
So with this change we always store both the `id`s and the `content_hash`s, right? We are no longer optimizing the case of caching by `id`, so that we only store the `id`s and not the latter?

Reply:
Right, so now they're being duplicated. 

Can we make it so that we don't?

Reply:
Nice thoughts about memory efficiency.
In our case, we don't duplicate the `doc_id` just the reference (`id(doc_id) == id(data)`).
`sys.getsizeof(doc_id) == 65` and the reference is just 8 bytes on 64bit systems.
We wouldn't save much memory and have to think if we would like to trade 8 bytes per document against more complexity. I think it would not be worth it for the cache.

Reply:
Ok, fair enough. But then the code and comments should be cleaned / removed / updated